components/formats-gpl/src/loci/formats/in/NDPISReader.java

```diff
@@ -102,14 +108,27 @@ public int getOptimalTileHeight() {
     readers[channel].setId(ndpiFiles[channel]);
     readers[channel].setSeries(getSeries());
     readers[channel].setResolution(getResolution());
-    int cIndex = channel < readers[channel].getSizeC() ? channel : 0;
-    int plane = readers[channel].getIndex(zct[0], cIndex, zct[2]);
-
-    readers[channel].openBytes(plane, buf, x, y, w, h);
-
-    return buf;
+    int spp = samplesPerPixel[channel];
+    if (spp==1) return readers[channel].openBytes(0, buf, x, y, w, h);    // single band reader
+
+    else {   // read intensity from used band (the other bands are close to zero, only jpeg artifacts and should be ignored)
+      byte[] bufReader = DataTools.allocate(w, h, spp); // w*h*RGB
+      bufReader = readers[channel].openBytes(0, bufReader, x, y, w, h);
+      int band = bandUsed[channel];
+      // each channel is RGB data (with usually only one band used), thus we sum up the intensities
+      if (readers[channel].isInterleaved()) {
+        for (int i = 0; i < buf.length; i++) {
+          buf[i] = bufReader[i * spp + band];
+        }
+      } else {    // not interleaved
+        final int offs = w * h;
+        System.arraycopy(bufReader,offs * band,buf,0,buf.length);
+      }
+      return buf;
+    }
```

Comment:
probably this would work, too. I'm not aware about the details of the ChannelSeparator.
(however, this solution seems to have less overhead / to be faster to me.)